### ISSUE TITLE
Can we use BART for next word prediction or autocompletion task?

### ISSUE THREAD

**Issue by vanh17** [pytorch/fairseq#1718]
I am trying to use BART for next word prediction like what BERT can do. Is there a way we could do this?

**Comment by lematt1991** [comment]
CC @ngoyal2707

**Comment by ngoyal2707** [comment]
@vanh17 Depends on what exact task you have.

1) In autocompletion typically, you have first `n` tokens and you need to predict `n+1`th token. For this, I don't expect `BERT` to be great because it never sees this type of data during pretraining. Either `UniLM (example: GPT2)`  or `BART` would work better. I have a feeling that similar sized `BART` model should outperform `UniLM` but haven't tested it.

2) Masked filling task: you have one or more tokens missing in the whole sentence. In this case, `BERT` should be good. `BART` should give you competing result also.

I haven't tried either myself but as I mentioned, both are fair things to try with `BART` model.

**Comment by vanh17** [comment]
I looked at the interface hub and see there is a function called generate. Is that a function we use to do next work prediction? If so how could we use such function for the task.
